src/compenentes/posts.js

```diff
@@ -0,0 +1,298 @@
+//funcion para enviar posts a base de datos
+addBtn.addEventListener('click', enviar);
+function enviar(){
+    db.collection("posts").add({
+        name : user.displayName,
+        texto : text.value,
+        likes: []
+    })
+    .then(function(docRef) {
+        console.log(docRef.id)
+    })
+    .catch(function(error) {
+        console.error("Error adding document: ", error);
+    });
+    document.querySelector('#txt').value = '';
+}
+
+
+db.collection("posts").onSnapshot((querySnapshot)=> {
+    root2.innerHTML ='';
+    querySnapshot.forEach((doc) => {
+        let post = `
+        <p>${doc.data().name}</p>
+        <p>${doc.data().texto}</p>
+        <button id="delete" value="${doc.id}" class="delete">Eliminar</button>
+        <button  id="edit" name="${doc.data().texto}" data-id="${doc.id}" class="edit">Editar</button>
+        <button class="like" name="${doc.data().likes}" data-id="${doc.id}"> Me gusta </button> <p class="like" >${doc.data().likes}</p>
+        `
+    let nodo = document.createElement('div');
+    nodo.classList.add('card')
+    nodo.innerHTML = post;
+    root2.appendChild(nodo);
+});
+//Boton para Eliminar
+let idbttn = document.querySelectorAll('.delete');
+    idbttn.forEach(btn => btn.addEventListener('click', (value) => {
+        value = btn.value;
+        db.collection("posts").doc(value).delete().then(function() {
+        }).catch(function(error) {
+          alert("Error removing document: ", error);
+        });
+}));
+ //Boton para contar Me Gusta
+ let like = document.querySelectorAll('.like');
+ like.forEach(likes=> likes.addEventListener('click', (e, iddoc)=>{
+     iddoc = likes.getAttribute('data-id');
+    let docRef =  db.collection("posts").doc(iddoc);
+    docRef.get().then(function(doc) {
+       if (doc.exists) {
+           console.log("Document data:", doc.data());
+           let like = doc.data().likes;
+           let increment = firebase.firestore.FieldValue.increment(1)
+           console.log(like)
+          return docRef.update({
+               name: doc.data().name,
+               texto: doc.data().texto,
+               likes: increment
+           })
+       } else {
+           // doc.data() will be undefined in this case
+           console.log("No such document!");
+       }
+   }).catch(function(error) {
+       console.log("Error getting document:", error);
+   });
+      
+ }))
+//Boton para Editar comentarios
+
+let editbttn = document.querySelectorAll('#edit');
+    editbttn.forEach(btnedit => btnedit.addEventListener('click', (valor, txt) => {
+        txt = btnedit.name; 
+        valor = btnedit.getAttribute('data-id');     
+          document.querySelector('#txt').value = txt;
+         document.querySelector('.guardar').style.display = 'block';
+        document.querySelector('.compartir').style.display = 'none';
+        let bttnsave = document.querySelector('.guardar');
+        bttnsave.addEventListener('click',function(){
```

Comment:
Evitemos escribir funciones anónimas dentro de ciclos [NADA DE LISTENERS VA DENTRO DE UN FOR]